### DISCUSSION TITLE
Q&A

### DISCUSSION THREAD

**Discussion by MyHomeMyData** [MyHomeMyData/ioBroker.e3oncan#26] [Announcements]
Place questions here.

Fragen gerne hier stellen.

**Comment by cb0b** [comment]
Wie kann ich denn den Schreibvorgang im ioBroker debuggen? Ich setze zB den Datenpunkt e3oncan.0.HPMUMASTER_0x680.json.0622_CustomerDetails klicke auf "set value" und das Ding bleibt "rot". Aktuellste Version aus dem Betarepo hab ich schon drauf. Die Anlage ist eine Vitocal 222-S via externem CANbus (91)

**Reply by MyHomeMyData** [reply]
Voraussetzung für Schreibzugriffe ist, dass ein "UDS-Worker" für das Device aktiv ist. D.h. Du musst mindestens einen Datenpunkt lesend in der Konfiguration für HPMUMASTER_0x680 festlegen.
Im Log/Protokoll des ioBroker solltest Du sehen, was passiert. Evtl. den Adapter vorher auf Log-Level "Debug" umstellen.

Habe es eben bei meiner Vitocal 250 probiert und konnte 622 problemlos schreiben.

**Reply by cb0b** [reply]
Danke - funktioniert astrein, Es geht jetzt mit allen Datenpunkten.

**Reply by MyHomeMyData** [reply]
Sehr schön, freut mich. Das mit dem aktiven UDS-Worker ist etwas blöd. Muss mal drüber nachdenken, wie ich das verbessern kann.

**Reply by MyHomeMyData** [reply]
Im Beta-Repo ist jetzt die Version 0.9.4 verfügbar. Damit wird das Problem behoben und Schreibvorgänge sind auf allen UDS-Devices ohne Vorbedingung möglich.

**Reply by cb0b** [reply]
Danke - habs gleich installiert. Was mir aufgefallen ist: nachdem ich den Scan beim setup gemacht habe werden alle Datenpunkte eingelesen. Der Timestamp im ioBroker ändert sich aber nicht mehr. Wenn ich für den Datenpunkt einen eigenen Request via UDSonCan anlege wird der Wert korrekt aktuallisiert. Ist das so gedacht?

**Reply by MyHomeMyData** [reply]
Wenn Du einen Datenpunktscan machst, sollten sämtliche Datenpunkte aller Geräte aktualisiert werden. Dabei müssten auch die Timestamps aktualisiert werden. Ich habe aber schon beobachtet, dass ioBroker die Timestamps manchmal nicht korrekt anzeigt. Erst wenn man die Seite neu lädt, werden alle Daten korrekt angezeigt. Ob das hier auch der Fall ist, kann ich natürlich nicht sagen.
Was wurde während des Scans im Log protokolliert?

Ein Datenpunktscan ist nach einem Updaten nur notwendig, wenn sich Definitionen der Datenpunkte geändert haben, oder neue dazu gekommen sind. Das ist dann im Changelog vermerkt und die geänderten Definitionen werden beim ersten Start des Adapters im Log protokolliert.

**Comment by cb0b** [comment]
Hab mal den Browser geschlossen und alle Kekse glöscht. Die Darstellung im ioBroker ändert sich dadurch nicht. 
![Screenshot 2024-08-27 101853](https://github.com/user-attachments/assets/eee2420b-644c-4ee6-a966-06344eef4905)
An der Uhrzeit im Screenshot kannst Du gut sehen was ich meine. Mindestens die müsste auf alle Fälle aktualisiert werden.
Der zweite Screenshot zeigt die DHW Temperatur die ich aktiv über UDSSonCAN auslese - die ist auch aktuell. Alle anderen drumherum nicht.
![Screenshot 2024-08-27 102706](https://github.com/user-attachments/assets/9ec69bda-a707-445c-bf45-a5b0fa3fda80)
Im log sehe ich die Aktualisierung vom DHW Wert (hab ich in der App geändert) leider nicht. Angezeigt wird die aber im ioBroker.

**Reply by MyHomeMyData** [reply]
Nach meinem Verständnis passt das, was ich da sehe. Im ersten Screenshot sieht man, dass alle Datenpunkte um 19:47:22 aufgefrischt wurden. Da lief vermutlich der Scan, denn der zeitlich Abstand ist jeweils im Bereich einiger 10 Millisekunden. Die ersten beiden Datenpunkte haben ihren Wert dabei nicht geändert, deshalb bleibt der Eintrag für die letzte Änderung bei 09:47:49 stehen.

Laut zweitem Screenshot wurde der DHW-Sollwert um um 10:26:44 durch den Adapter aufgefrischt, hat sich aber ggü. dem Wert von 19:47:18 am Vortag nicht geändert. Im Log werden keine Lese-Vorgänge protokolliert, das kann dort also nicht erscheinen. Da Du den Wert in der App geändert hast, hat der Adapter ja nichts geschrieben. Wenn die Aktualisierung in ioBroker angezeigt wird, ist doch alles ok, oder?

**Comment by cb0b** [comment]
Nee das passt leider nicht. Der DHW Wert wird ja aktiv über USDonCAN ausgelesen und auch tatsächlich aktualisiert - alles fein. Wenn ich den aktiven Scan aber wegnehme wird der Wert nicht aktualisiert (über die vicare App Wert verstellt). Wo man das sehr gut sieht ist beim Datenpunkt alias.0.vitocal.RO.0271_DomesticHotWaterSensor (siehe Screenshot) Das Warmwasser hat aktuell 11:29 27.8.24 42°C.
![Screenshot 2024-08-27 112655](https://github.com/user-attachments/assets/888a7df3-b010-4475-9552-c7c11ebb0d56)

**Reply by cb0b** [reply]
Könnte das daran liegen dass ich nur den externen CanBus via Adapter am Raspi habe?

**Reply by MyHomeMyData** [reply]
Meinst Du mit "Wenn ich den aktiven Scan aber wegnehme", dass keine UDS-Schedule für das Lesen des Datenpunktes mehr aktiv ist?

In dem Fall kann der Wert nicht aktualisiert werden, weil die Vitocal von sich aus die Daten nur auf dem **internen CAN** sendet. Ein "Collect" auf dem externen CAN funktioniert da leider nicht, weil da keine Daten der Vitocal ankommen.

**Reply by cb0b** [reply]
Prima - das heisst ich habe jetzt zwei Optionen:
1: ich klemme den Adapter auf den internen CAN Bus und bekomme so alle Änderungen automatisch mit oder
2: ich mache mir für jeden Datenpunkt den ich aktualisiert haben will eine eigene Schedule via UDSonCAN auf dem externen CANbus
Is das korrekt so?

**Reply by MyHomeMyData** [reply]
Jetzt nochmal zusammengefasst an der richtigen Stelle:

Du kannst alle Datenpunkte in eine Schedule packen, mit Komma getrennt. Collect brauchst Du dann nicht. Natürlich kannst Du auch mehrere Schedules machen, z.B. für Messwerte alle 30 Sekunden, für kumulierte Werte alle 5 Minuten.

Alternativ kannst auch auf den internen Bus umklemmen und solltest damit die meisten Datenpunkte per "Collect" einsammeln können. Das funktioniert aber nicht mit allen Datenpunkten. Sollwerte werden nach meiner Erfahrung von der Vitocal nicht zuverlässig versendet. Das gilt auch für einige andere Punkte. Du wirst also zusätzlich auch Punkte per UDSonCAN Schedule abfragen wollen. Das sollte auch über den internen Bus funktionieren - das musst Du ausprobieren, ich habe bisher nur Erfahrung damit auf dem externen Bus.

Vielleicht schaust Du Dir auch meine [Beschreibung der Topologien](https://github.com/MyHomeMyData/ioBroker.e3oncan/discussions/34) an.

**Reply by cb0b** [reply]
Danke für die Hilfe. Ich werd das dann genauso machen wie Du es beschreibst und ich muss den Adapter nicht umbauen. :-)
